src/lavinmq/http/controller/queues.cr

```diff
@@ -41,6 +42,25 @@ module LavinMQ
           end
         end
 
+        get "/api/queues/:vhost/:name/unacked" do |context, params|
+          with_vhost(context, params) do |vhost|
+            refuse_unless_management(context, user(context), vhost)
+            unacked_messages = Array(UnackedMessage).new
+            @amqp_server.vhosts[vhost].queues[params["name"]].@consumers.each do |c|
+              c.unacked_messages.select { |u| u.queue.name == params["name"] }.each do |u|
```

Comment:
take care to not create arrays if we don't need to: 

```suggestion
              c.unacked_messages.each.select { |u| u.queue.name == params["name"] } do |u|
```

Reply:
Wait, that doesn't work, we need the each at the end for the loop? I removed the select and added a next in the loop instead. 
```
c.unacked_messages.each do |u|
  next unless u.queue.name == params["name"]
```

Reply:
I guess we can get rid of `unacked_messages = Array(UnackedMessage).new` too by making sure everything is iterators?